glean-core/src/upload/directory.rs

```diff
@@ -151,29 +178,40 @@ impl PingDirectoryManager {
         None
     }
 
-    /// Process the pings directory and return a vector with the ping data
+    /// Process both ping directories.
+    pub fn process_dirs(&self) -> PingPayloadsByDirectory {
+        PingPayloadsByDirectory {
+            pending_pings: self.process_dir(&self.pending_pings_dir),
+            deletion_request_pings: self.process_dir(&self.deletion_request_pings_dir),
+        }
+    }
+
+    /// Process one of the pings directory and return a vector with the ping data
     /// corresponding to each valid ping file in the directory.
     /// This vector will be ordered by file `modified_date`.
     ///
     /// Any files that don't match the UUID regex will be deleted
     /// to prevent files from polluting the pings directory.
     ///
-    /// Files that are not correctly formatted will also be deleted.
-    ///
     /// # Return value
     ///
-    /// `Vec<(String, String, JsonValue, HeaderMap)>` -
-    ///     a vector of tuples containing the document_id, path, body and headers of each request.
-    pub fn process_dir(&self) -> Vec<PingPayload> {
+    /// `Vec<(fs::Metadata, PingPayload)>` -
+    ///     a vector of tuples with the file metadata and payload of each ping file in the directory.PathBuf
+    fn process_dir(&self, dir: &PathBuf) -> Vec<(fs::Metadata, PingPayload)> {
         log::info!("Processing persisted pings.");
 
-        // Walk the pings directory and process each file in it,
-        // deleting invalid ones and ignoring unreadable ones.
-        // Create a vector of tuples: (modified_date, PingRequest)
-        // using the contents and metadata of all valid files.
-        let mut pending_pings: Vec<_> = self
-            .get_ping_entries()
-            .into_iter()
+        let entries: fs::ReadDir;
+        match dir.read_dir() {
+            Ok(r) => entries = r,
+            Err(_) => {
+                // This may error simply because the directory doesn't exist,
+                // which is expected if no pings were stored yet.
+                return Vec::new();
+            }
+        }
```

Comment:
`match` is an expression and returns a value, we can therefore do this:

```suggestion
        let entries = match dir.read_dir() {
            Ok(entries) => entries,
            Err(_) => {
                // This may error simply because the directory doesn't exist,
                // which is expected if no pings were stored yet.
                return Vec::new();
            }
        };
```